src/main/java/com/org/checkr/controller/CandidatesController.java

```diff
@@ -28,7 +34,16 @@ public List<CandidateDTO> getAllCandidates(@RequestParam(name = "name", required
 
     }
     @PostMapping
-    public Candidate createCandidate(@RequestBody Candidate candidate) {
-        return candidateService.saveCandidate(candidate);
+    public ResponseEntity<CompleteCandidateInfoDTO> createCandidate(@RequestBody CreateCandidateRequestDTO candidateRequestDTO) {
+        Candidate candidate = candidateService.createCandidate(candidateRequestDTO);
+        CompleteCandidateInfoDTO responseDto = candidateService.mapCandidateEntityToDto(candidate);
+        return new ResponseEntity<>(responseDto, HttpStatus.CREATED);
+    }
+
+    @PatchMapping("/{id}")
+    public ResponseEntity<CompleteCandidateInfoDTO> engageWithCandidate(@PathVariable Long id) {
+        Candidate candidate = candidateService.engageWithCandidate(id);
+        CompleteCandidateInfoDTO responseDto = candidateService.mapCandidateEntityToDto(candidate);
+        return new ResponseEntity<>(responseDto, HttpStatus.OK);
```

Comment:
what happen if engageWithCandidate returns null or not able to complete. Please check if we can add another status code if there is error in processing

Reply:
added new status code incase of null